var/spack/repos/builtin/packages/grass/package.py

```diff
@@ -0,0 +1,176 @@
+
+    variant('cxx',       default=True,  description='Add c++ functionality')
+    variant('tiff',      default=True,  description='Add TIFF functionality')
+    variant('png',       default=True,  description='Add PNG functionality')
+    variant('postgres',  default=False, description='Add PostgreSQL functionality')
+    variant('mysql',     default=False, description='Add MySQL functionality')
+    variant('sqlite',    default=True,  description='Add SQLite functionality')
+    variant('opengl',    default=True,  description='Add OpenGL functionality')
+    variant('fftw',      default=True,  description='Add FFTW functionality')
+    variant('blas',      default=False, description='Add BLAS functionality')
+    variant('lapack',    default=False, description='Add LAPACK functionality')
+    variant('cairo',     default=True,  description='Add Cairo functionality')
+    variant('freetype',  default=True,  description='Add FreeType functionality')
+    variant('readline',  default=False, description='Add Readline functionality')
+    variant('regex',     default=True,  description='Add regex functionality')
+    variant('pthread',   default=False, description='Add POSIX threads functionality')
+    variant('openmp',    default=False, description='Add OpenMP functionality')
+    variant('opencl',    default=False, description='Add OpenCL functionality')
+    variant('bzlib',     default=False, description='Add BZIP2 functionality')
+    variant('netcdf',    default=False, description='Enable NetCDF support')
+    variant('geos',      default=False, description='Geometry Engine for v.buffer')
+
+    # required components
+    depends_on('gmake@3.8.1:', type='build')
+    depends_on('zlib')
+    depends_on('flex')
+    depends_on('bison')
+    depends_on('proj')
+    depends_on('gdal')
+    depends_on('python@2.7:2.99')
+    depends_on('libx11')
+
+    # optional pieces
+    depends_on('libtiff', when='+tiff')
+    depends_on('libpng', when='+png')
+    depends_on('postgresql', when='+postgres')
+    depends_on('mariadb', when='+mysql')
+    depends_on('sqlite', when='+sqlite')
+    depends_on('opengl', when='+opengl')
+    depends_on('fftw', when='+fftw')
+    depends_on('blas', when='+blas')
+    depends_on('lapack', when='+lapack')
+    depends_on('cairo', when='+cairo')
+    depends_on('freetype', when='+freetype')
+    depends_on('readline', when='+readline')
+    depends_on('bzip2', when='+bzlib')
+    depends_on('netcdf', when='+netcdf')
+    depends_on('geos', when='+geos')
+
+    def configure_args(self):
+        spec = self.spec
+
+        args = [
+            '--without-odbc',
+            '--without-nls',
+            '--without-opendwg',
+            '--with-x',
+            '--with-gdal={0}/bin/gdal-config'.format(
+                spec['gdal'].prefix),
+        ]
+
+        if '-cxx' in spec:
+            args.append('--without-cxx')
+
+        if '-tiff' in spec:
+            args.append('--without-tiff')
```

Comment:
I like to be explicit and say:
```python
if '+tiff' in spec:
    args.append('--with-tiff')
else:
    args.append('--without-tiff')
```
for all dependencies. Otherwise, a lot of build systems will build optional support if they can find the dependency on the OS. We don't want these kind of non-deterministic builds in Spack.

Reply:
+tiff is the default, which will load up libtiff.  Mine is not as symmetrical as it follows the configuration file - which probably will change at any time, so your way is better.